### PR TITLE
Add validations - minLength, maxLength, isEqualTo

### PR DESCRIPTION
**Pull request by groces**
Added the following validations:
- `minLength`
- `maxLength`
- `isEqualTo`

Added unit tests

### CODE REVIEW

File: lib/check.js [hunk 1]
```diff
@@ -192,6 +195,30 @@ const defined = basic( _isNot(undefined), MSG.REQUIRED)
...
+)(text))
+
+// maxLength = Integer -> String -> Maybe String
+const maxLength = basic(_isLessThanMax, MSG.MAXIMUM_LENGTH_EXCEEDED)
+
+// _isEqualTo = String -> Value -> Object -> Maybe String
+const _isEqualTo = _.curry((key, value, obj) => _.propEq(key, value)(obj))
+
+// isEqualTo = String -> Value, Object -> Maybe String
+const isEqualTo = basic(_isEqualTo, MSG.DOES_NOT_MATCH)
```
Comment (scull7) [inline lib/check.js]:
```javascript
const isEqualTo = _.compose(
  _.curryN(3)
, basic(_.propEq, MSG.DOES_NOT_MATCH)
)
```

**Comment by mszymulanski** [comment]
I think you can accomplish the same results by using `isInRangeInclusive`.

**Comment by scull7** [comment]
@mszymulanski to which function are you referring?

**Comment by mszymulanski** [comment]
Sry,  I was thinking about `isLength`. I was referring to minLength abd maxLength.  I also dont like naming in those functions as they can be generic and not reserved only for text (String). 

Reviewing from a phone sucks

File: lib/check.js (hunk 1 above)
Comment (scull7) [inline lib/check.js]: `basic` needs to be changed: https://goo.gl/zrnofd
```javascript
const basic = (test, msg) => (...args) => {
  return test.apply(null, args) ? null : msg
}
const isEqualTo = R.curryN(3, basic(R.propEq, "FAIL"))

const request = { foo: 'bar' }

isEqualTo('foo', 'bar', request)
```

**Pull request merged**